### PR TITLE
Kafka Implementation

### PR DESCRIPTION
**Pull request by ojahagirdar**
-  implement Batch Processing 

-  implement Producer Configuration

-  implement  Consumer Configuration

-add service layer for Producer  and Consumer 

-kafka unit test

### CODE REVIEW

File: src/main/java/com/solr/clientwrapper/rest/KafkaResource.java [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+@RestController
+@RequestMapping("/kafka")
+public class KafkaResource {
+
+	private final Logger log = LoggerFactory.getLogger(KafkaResource.class);
+
+	@Autowired
+	private KafkaSender sender;
+
+	@PostMapping("/Add/{AddToQueue}")
```
Comment (skdhitman) [inline src/main/java/com/solr/clientwrapper/rest/KafkaResource.java:27]: Best practice is to use **_hyphenated words_** instead of SnakeCase or camelCase in _APIs URIs_

File: src/main/java/com/solr/clientwrapper/domain/port/api/kafkaSender.java [hunk 2]
```diff
@@ -0,0 +1,7 @@
+package com.solr.clientwrapper.domain.port.api;
+
+public interface kafkaSender {
+	
+	public String AddToQueue(String payload);
```
Comment (skdhitman) [inline src/main/java/com/solr/clientwrapper/domain/port/api/kafkaSender.java:5]: camelCase naming convention should be followed for method names

**Review by skdhitman — COMMENTED** [review]
Suggestion:
A little typo: _receive_
You can consider changing it if you would like to.

File: src/main/java/com/solr/clientwrapper/domain/port/api/kafkaSender.java (hunk 2 above)
Comment (karthik261099) [reply src/main/java/com/solr/clientwrapper/domain/port/api/kafkaSender.java:5]: Camel case - "addToQueue"

File: src/main/java/com/solr/clientwrapper/domain/port/api/kafkaSender.java [hunk 3]
```diff
@@ -0,0 +1,7 @@
+package com.solr.clientwrapper.domain.port.api;
+
+public interface kafkaSender {
```
Comment (karthik261099) [inline src/main/java/com/solr/clientwrapper/domain/port/api/kafkaSender.java:3]: Class/interface names should start with capital letter

File: src/main/java/com/solr/clientwrapper/domain/service/KafkaSender.java [hunk 4]
```diff
@@ -0,0 +1,40 @@
...
+	private static final Logger LOGGER = LoggerFactory.getLogger(KafkaSender.class);
+
+	@Autowired
+	private KafkaTemplate<String, String> kafkaTemplate;
+
+	@Value("${kafka.topic-name}")
+	private String topicName;
+
+	@Override
+	public String AddToQueue(String payload) {
```
Comment (karthik261099) [inline src/main/java/com/solr/clientwrapper/domain/service/KafkaSender.java:27]: camel case

File: src/main/java/com/solr/clientwrapper/domain/service/SolrDocumentService.java [hunk 5]
```diff
@@ -0,0 +1,238 @@
...
+import org.springframework.beans.factory.annotation.Value;
+import org.springframework.kafka.core.KafkaTemplate;
+import org.springframework.stereotype.Service;
+import org.springframework.transaction.annotation.Transactional;
+
+import java.util.*;
+
+@Service
+@Transactional
+public class SolrDocumentService implements SolrDocumentServicePort {
```
Comment (karthik261099) [inline src/main/java/com/solr/clientwrapper/domain/service/SolrDocumentService.java:24]: Why did you add my code here?

File: src/main/java/com/solr/clientwrapper/rest/KafkaResource.java (hunk 1 above)
Comment (karthik261099) [reply src/main/java/com/solr/clientwrapper/rest/KafkaResource.java:27]: add-to-queue

File: src/main/java/com/solr/clientwrapper/rest/SolrInputDocumentResource.java [hunk 6]
```diff
@@ -0,0 +1,43 @@
...
+import io.swagger.v3.oas.annotations.security.SecurityRequirement;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.http.HttpStatus;
+import org.springframework.http.ResponseEntity;
+import org.springframework.web.bind.annotation.*;
+
+@RestController
+@RequestMapping("/document")
+public class SolrInputDocumentResource {
```
Comment (karthik261099) [inline src/main/java/com/solr/clientwrapper/rest/SolrInputDocumentResource.java:15]: Why other branch's code is added here?

File: src/main/java/com/solr/clientwrapper/usecase/solr/document/CreateSolrDocument.java [hunk 7]
```diff
@@ -0,0 +1,27 @@
...
+import com.solr.clientwrapper.domain.dto.solr.SolrResponseDTO;
+import com.solr.clientwrapper.domain.port.api.SolrDocumentServicePort;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.stereotype.Service;
+import org.springframework.transaction.annotation.Transactional;
+
+@Service
+@Transactional
+public class CreateSolrDocument {
```
Comment (karthik261099) [inline src/main/java/com/solr/clientwrapper/usecase/solr/document/CreateSolrDocument.java:12]: Other branch's code

File: src/main/resources/config/application-dev.yml [hunk 8]
```diff
@@ -1,112 +1,114 @@
...
-# ===================================================================
-
-# ===================================================================
-# Standard Spring Boot properties.
-# Full reference is available at:
-# http://docs.spring.io/spring-boot/docs/current/reference/html/common-application-properties.html
-# ===================================================================
-   
-
-logging:
```
Comment (karthik261099) [inline src/main/resources/config/application-dev.yml:17]: any reason to delete all these configs in the file and add it again?

File: src/main/java/com/solr/clientwrapper/rest/KafkaResource.java (hunk 1 above)
Comment (ojahagirdar) [reply src/main/java/com/solr/clientwrapper/rest/KafkaResource.java]: done

File: src/main/java/com/solr/clientwrapper/domain/port/api/kafkaSender.java (hunk 2 above)
Comment (ojahagirdar) [reply src/main/java/com/solr/clientwrapper/domain/port/api/kafkaSender.java]: done

File: src/main/java/com/solr/clientwrapper/domain/port/api/kafkaSender.java (hunk 3 above)
Comment (ojahagirdar) [reply src/main/java/com/solr/clientwrapper/domain/port/api/kafkaSender.java]: done

File: src/main/java/com/solr/clientwrapper/domain/service/KafkaSender.java (hunk 4 above)
Comment (ojahagirdar) [reply src/main/java/com/solr/clientwrapper/domain/service/KafkaSender.java]: done
